File: Week01/Decimal_to_Binary/Bitwise.cs
using static System.Console;


class Bitwise
{
    static string decToBinary(int n)
    {
        StringBuilder bin = new StringBuilder();

        while (n > 0)
        {
            int bit = n & 1;
            bin.Append(bit);

            n = n >> 1;
        }

        char[] arr = bin.ToString().ToCharArray();
        Array.Reverse(arr);
        return new string(arr);
    }


    static void Main()
    {
        int n = 12;
        WriteLine(decToBinary(n));
    }
}

File: Week01/Decimal_to_Binary/Bitwise.cpp
#include<iostream>
#include<algorithm>

using namespace std;


string decToBinary(int n) {
    string bin = "";

    while (n > 0) {
        int bit = n & 1;
        bin.push_back('0' + bit);

        n = n >> 1;
    }

    reverse(bin.begin(), bin.end());
    return bin;
}


int main() {
    int n = 12;
    cout << decToBinary(n);
    return 0;
}